dangerzone/document.py

```diff
@@ -18,13 +22,26 @@ class Document:
     document, and validating its info.
     """
 
+    doc_counter = 1
```

Comment:
Mutable state at class level is usually prone to bugs. It's much better to assign an alphanumerical ID (can be just a few characters) that will uniquely track the document instance.

Reply:
But then what do you print in the output? I am printing something like this at the moment:

```
[doc 11] 0% Converting to PDF using LibreOffice
[doc 5] 68% Converting page 3/5 from pixels to PDF
[doc 5] 77% Converting page 4/5 from pixels to PDF
[doc 12] 0% Converting to PDF using LibreOffice
[doc 5] 86% Converting page 5/5 from pixels to PDF
[doc 5] 95% Merging 5 pages into a single PDF
[doc 6] 100% Safe PDF created
```

With alphanumeric values it's going to look nonsensical, no?

Reply:
I guess something like this:

```
[doc b1am3] Assigning ID 'b1ame' to doc C:\Users\dz\doc5.pdf
[doc 0rgs3] Assigning ID 'orgs3' to doc C:\Users\dz\doc6.pdf
[doc 9ein9] Assigning ID '9ein9' to doc C:\Users\dz\doc11.pdf
[doc c1trus] Assigning ID 'c1trus' to doc C:\Users\dz\doc12.pdf
...
[doc 9ein9] 0% Converting to PDF using LibreOffice
[doc b1am3] 68% Converting page 3/5 from pixels to PDF
[doc b1am3] 77% Converting page 4/5 from pixels to PDF
[doc c1trus] 0% Converting to PDF using LibreOffice
[doc b1am3] 86% Converting page 5/5 from pixels to PDF
[doc b1ame] 95% Merging 5 pages into a single PDF
[doc 0rgs3] 100% Safe PDF created
```

_(I'm adding the association between IDs and filenames, which we need to do regardless)_

As for the fact that they are alphanumerical, I think that's a bonus, as you can easily grep for them in the logs.

Reply:
I like the alignment in the log that this will bring and the searchability bonus. Some users might be confused and think that's like a commit hash. But I think it's OK.